src/toil/wdl/wdltoil.py

```diff
@@ -1979,15 +1984,23 @@ def run(self, file_store: AbstractFileStore) -> Promised[WDLBindings]:
 
                 per_part_size = convert_units(part_size, part_suffix)
                 total_bytes += per_part_size
-                if specified_mount_point is not None:
-                    if mount_spec.get(specified_mount_point) is not None:
+                if mount_spec.get(specified_mount_point) is not None:
```

Comment:
This check doesn't account for how `local-disk` and no mount point mean the same thing, so will still let you specify both.

If there was a test for this validation logic, that would catch the problem.

Reply:
I mapped `local-disk` and no mount point to the same thing in a conditional branch below. I changed the logic a bit so it's more clear and so there's only one conditional branch that does this duplication check.